apps/build-onchain-apps/src/contract/BuyMeACoffee.ts

```diff
@@ -114,6 +112,6 @@ const abi: Abi = [
     type: 'function',
   },
   { stateMutability: 'payable', type: 'receive' },
-];
+] as const;
```

Comment:
See: https://wagmi.sh/react/prepare-hooks/usePrepareContractWrite#abi-optional

We should declare abis `as const` and just let the types be inferred. If we define them as the `Abi` type, it's too broad and we don't get good typechecking. 

With this approach we get nice autocomplete and errors on `functionName`, for example.

<img width="842" alt="image" src="https://github.com/coinbase/build-onchain-apps/assets/9300702/968c5ddc-7274-4348-8b01-c488e65cc552">